pulumi-language-dotnet/main.go

```diff
@@ -547,8 +559,42 @@ func (w *logWriter) LogToUser(val string) (int, error) {
 	return len(val), nil
 }
 
+func (host *dotnetLanguageHost) buildDll() error {
+	// If we are running from source, we need to build the project.
+	// Run the `dotnet build` command.  Importantly, report the output of this to the user
+	// (ephemerally) as it is happening so they're aware of what's going on and can see the progress
+	// of things.
+	args := []string{"build", "-nologo", "-o", "bin/pulumi-debugging"}
+
+	cmd := exec.Command(host.exec, args...)
+	err := cmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "failed to build project: %v", err)
+	}
+
+	err = filepath.WalkDir(".", func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if name, ok := strings.CutSuffix(d.Name(), ".csproj"); ok {
+			host.binary = filepath.Join("bin", "pulumi-debugging", name+".dll")
```

Comment:
This probably shouldn't change global state. A language plugin can be used for multiple things in parallel and this is gonna confuse things.
Honestly "binary" shouldn't even be a global var to start with, the host should be re-parsing it from options on each run.

Reply:
Changed that now.  I haven't fixed the global `host.binary` though, that's probably something to look at in a separate PR.